stac_geoparquet/stac_geoparquet.py

```diff
@@ -46,6 +55,17 @@ def to_geodataframe(items: Sequence[dict[str, Any]]) -> geopandas.GeoDataFrame:
             if k in item2:
                 raise ValueError("k", k)
             item2[k] = v
+        if add_self_link:
+            self_href = None
+            for link in item["links"]:
+                if (
+                    link["rel"] == "self"
+                    and (not link["type"] or link["type"] in STAC_ITEM_TYPES)
+                    and urlparse(link["href"]).netloc
```

Comment:
I also wonder about this. To me, it'd be a bit surprising to have `self` links not show up when I set `add_self_link=True`. I can imagine use-cases where a user might want to expand a relative link and then make it absolute later on.

Reply:
It's checking whether a self link is absolute. A relative self link seems not very useful in this context as the folder structure is gone (and it's actually forbidden in STAC, but people still sometime provide relative self links).